test/wrappers/ortools_test.rb

```diff
@@ -5777,4 +5777,26 @@ def test_pause_should_be_last_if_possible
   def test_no_nil_in_corresponding_mission_ids
     assert_empty OptimizerWrapper.config[:services][:ortools].send(:corresponding_mission_ids, ['only_id'], ['non_id'])
   end
+
+  def test_self_selection_should_not_change_vehicle_ids
+    vrp = VRP.independent_skills
+    vrp[:vehicles].each_with_index{ |v, v_i|
+      v.delete(:skills)
+      next if v_i.zero?
+
+      v[:skills] = [['compatible_skill']] # avoid using first vehicle
+    }
+    vrp[:services].each{ |s| s[:skills] = ['compatible_skill'] }
+    vrp[:configuration][:preprocessing] = { first_solution_strategy: ['self_selection'] }
+
+    Interpreters::SeveralSolutions.stub(
+      :collect_heuristics,
+      lambda { |_, _|
+        %w[parallel_cheapest_insertion global_cheapest_arc local_cheapest_insertion savings]
+      }
+    ) do
+      service_vrp = Interpreters::SeveralSolutions.find_best_heuristic({ service: :ortools, vrp: TestHelper.create(vrp) })
+      assert_equal 3, service_vrp[:vrp].vehicles.collect(&:id).uniq.size, 'We expect same number of IDS as initial vehicles in problem'
```

Comment:
Can we also stub the `OptimizerWrapper.solve` and return empty result to skip solution process altogether?

Reply:
If we return empty result that will not generate the error. We need one non empty route. We could eventually stub and return always same solution but this test needs 0.28s, I don't think we need to spend more time on it.

Reply:
Knowing what we test here, it shouldn't even take 0.01 seconds. I think your idea, returning a fixed solution, is a good idea in fact.